src/declarations/stencil-public-runtime.ts

```diff
@@ -1364,6 +1374,7 @@ export namespace JSXBase {
     tabIndex?: number;
     tabindex?: number | string;
     title?: string;
+    popover?: 'auto' | 'manual';
```

Comment:
So, this deviates a bit from the reproduction case in the linked issue and the [HTML spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/popover) for the global `popover` attribute. According to the spec, the `popover` attribute will default to "auto" if you do something like this:

```html
<div popover>...</div>
```

But, if we update our declaration here to be:
```ts
popover?: true | 'auto' | 'manual';
```
and write the same HTML, we actually end up getting the behavior for the "manual" value because it does not reflect the attribute correctly.

Not sure if this is something we've encountered before or can get around, but wanted to call it out since with this definition you'll be forced to explicitly set "auto" or "manual".

Reply:
should we maybe leave a jsx comment here to that effect?

Reply:
+1 for a comment here.

Looking at `lib.dom.ts`, it gets typed as:
```
    popover: string | null;
```
I assume they use `string` here rather than `'auto' | 'manual'` to:
1. allow for additional values to be added to [the spec ](https://html.spec.whatwg.org/multipage/popover.html#the-popover-attribute)
2. adhere to this part of the spec: "The [popover](https://html.spec.whatwg.org/multipage/popover.html#attr-popover) attribute's [invalid value default](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#invalid-value-default) is the [manual](https://html.spec.whatwg.org/multipage/popover.html#attr-popover-manual-state) state..."

In the case of `null`, I think that's to handle the latter part of the same sentence in the spec (emphasis mine):
> The [popover](https://html.spec.whatwg.org/multipage/popover.html#attr-popover) attribute's [invalid value default](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#invalid-value-default) is the [manual](https://html.spec.whatwg.org/multipage/popover.html#attr-popover-manual-state) state and i**ts [missing value default](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#missing-value-default) is the no popover state.**

Thoughts on aligning this with `lib.dom.ts`' typings?

Reply:
We can do that. Thoughts on typing it as `'auto' | 'manual' | string | null`? I know that's different from the spec a bit, but offers a bit of intellisense for devs. But, I also get if we don't wanna do that in case they remove one of those values. Any preference?

Reply:
I think that's fine - I doubt one of those will be removed, so the only thing I wanna watch out for is making sure `string` there doesn't widen "over" the `auto | manual` (basically ensure that the intellisense works for those value).

Reply:
`string` did widen over the other types so I just ended up matching the `lib.dom.ts` types.